app/src/main/java/com/example/weather/CityPickerAdapter.kt

```diff
@@ -1,52 +1,60 @@
-import com.example.weather.R
+import com.example.weather.databinding.CityItemBinding
+import com.example.weather.network.City
 
 /**
  * Adapter for the [RecyclerView] in [CityPickerFragment].
  */
 
-class CityPickerAdapter(private val dataSet: List<Int>, private val context: Context?) :
+class CityPickerAdapter(private val dataSet: LiveData<List<City>>, private val context: Context?) :
     RecyclerView.Adapter<CityPickerAdapter.ViewHolder>() {
 
     /**
      * Provide a reference to the type of views that you are using
      * (custom ViewHolder)
      */
-    class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
-        val textView: TextView
-
-        init {
-            // Define click listener for the ViewHolder's View
-            textView = view.findViewById(R.id.cityTextView)
-        }
-    }
+    class ViewHolder(val binding: CityItemBinding) : RecyclerView.ViewHolder(binding.root)
 
     // Create new views (invoked by the layout manager)
-    override fun onCreateViewHolder(viewGroup: ViewGroup, viewType: Int): ViewHolder {
-        // Create a new view, which defines the UI of the list item
-        val view = LayoutInflater.from(viewGroup.context)
-            .inflate(R.layout.city_item, viewGroup, false)
-
-        return ViewHolder(view)
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
+        val inflater = LayoutInflater.from(parent.context)
+        val binding = CityItemBinding.inflate(inflater, parent, false)
+        return ViewHolder(binding)
     }
 
     // Replace the contents of a view (invoked by the layout manager)
     override fun onBindViewHolder(viewHolder: ViewHolder, position: Int) {
 
         // Get element from your dataset at this position and replace the
         // contents of the view with that element
-        viewHolder.textView.text = context?.resources?.getString(R.string.list_item, position+1)
+        val dataSetValues: List<City>? = dataSet.value
+        viewHolder.binding.name.text = dataSetValues?.get(position)?.name.toString()
+        viewHolder.binding.country.text = dataSetValues?.get(position)?.sys?.country.toString()
+        viewHolder.binding.temp.text = context?.resources?.getString(R.string.temp, convertCelsiusToFahrenheit(dataSetValues?.get(position)?.main?.temp).toString())
+        viewHolder.binding.lowAndHigh.text = formatLowAndHigh(dataSetValues, position)
+        viewHolder.binding.humidity.text = context?.resources?.getString(R.string.humidity, dataSetValues?.get(position)?.main?.humidity.toString())
+    }
+
+    private fun formatLowAndHigh(dataSetValues: List<City>?, position: Int): String {
+        val low = convertCelsiusToFahrenheit(dataSetValues?.get(position)?.main?.temp_min).toString()
+        val high = convertCelsiusToFahrenheit(dataSetValues?.get(position)?.main?.temp_max).toString()
+        return "$low\u2109/$high\u2109"
+    }
+
+    private fun convertCelsiusToFahrenheit(temp: Double?): Int {
```

Comment:
let's try setting the units to "imperial" and seeing if that works. Then we can get rid of this extra code